docs/technical-details/reference/graphql.md

```diff
@@ -0,0 +1,235 @@
+
+For example, the variable `$name: String!` states that `name` should be a non-null, string value.
+- If the request does not contain the `name` variable, an error will be returned
+- If the request provides `null` for the `name` variable, an error will be returned
+- If the request provides any non-string value for the `name` variable, i.e. `1`, `true`, `{ foo: "bar" }`, the behavior is undefined and an error may or may not be returned.
+- If the variable definition is changed to include a default value, `$name: String! = "John"`, then when omitted, `"John"` will be used.
+  - If `null` is provided as the variable value, an error will still be returned.
+  - If the default value does not match the type specified (i.e. `$name: String! = 0`), this is also considered undefined behavior. It may or may not fail in a variety of ways.
+- Fragments will generally extend non-specified types, and the querying system will do no validity checking on them. For example, `fragment Fields on Any { ... }` is just as valid as `fragment Fields on MadeUpTypeName { ... }`. See the Fragments sections for more details.
+
+The only notable place the querying system will do some level of type analysis is the transformation of arguments into a query.
+- Objects will be transformed into properly nested attributes
+- Strings and Boolean values are passed through as their AST values
+- Float and Int values will be parsed using the JavaScript `parseFloat` and `parseInt` methods respectively.
+- List and Enums are not supported.
+
+### Fragments
+
+The querying system loosely supports fragments. Both fragment definitions and inline fragments are supported, and are entirely a composition utility. Since this system does very little type checking, the `on Type` part of fragments is entirely pointless. Any value can be used for `Type` and it will have the same effect.
+
+For example, in the query
+
+```graphql
+query Get {
+	Dog {
+		...DogFields
+	}
+}
+
+fragment DogFields on Dog {
+	name
+	breed
+}
+```
+
+The `Dog` type in the fragment has no correlation to the `Dog` resource in the query (that correlates to the HarperDB `Dog` resource).
+
+You can literally specify anything in the fragment and it will behave the same way:
+
+```graphql
+fragment DogFields on Any { ... } # this is recommended
+fragment DogFields on Cat { ... }
+fragment DogFields on Animal { ... }
+fragment DogFields on LiterallyAnything { ... }
+```
+
+As an actual example, fragments should be used for composition:
+
+```graphql
+query Get {
+	Dog {
+		...sharedFields
+		breed
+	}
+	Owner {
+		...sharedFields
+		occupation
+	}
+}
+
+fragment sharedFields on Any {
+	id
+	name
+}
+```
+
+### Short Form Querying
+
+Any Resource attribute can be used as an argument for a query. In this short form, multiple arguments is treated as multiple equivalency conditions with the default `and` operation.
```

Comment:
Is a resource attribute different from a table attribute?

Reply:
No, since a table is a resource. However, it may be less likely to confuse to just say "Any attribute", or "Any field".

Reply:
Fixed